src/constraint/SequentialSampleable.cpp

```diff
@@ -0,0 +1,137 @@
+#include "aikido/constraint/SequentialSampleable.hpp"
+
+#include <dart/common/StlHelpers.hpp>
+
+namespace aikido {
+namespace constraint {
+
+//==============================================================================
+class SequentialSampleGenerator : public SampleGenerator
+{
+public:
+  SequentialSampleGenerator(
+      statespace::StateSpacePtr stateSpace,
+      std::vector<std::unique_ptr<SampleGenerator>> generators)
+    : mStateSpace(std::move(stateSpace))
+    , mGenerators(std::move(generators))
+    , mIndex(0)
+  {
+#ifndef NDEBUG
+    for (const auto& generator : mGenerators)
+      assert(generator->getStateSpace() == mStateSpace);
```

Comment:
Nit: Could we add an assertion for checking nullity of the generators?

Reply:
@jslee02  Should it be an _assertion_? :thinking: